RPC.en.rst

```diff
@@ -3,59 +3,257 @@
 .. highlight:: cpp
 .. default-domain:: cpp
 
-.. _rpc_refactor:
+.. _rpc:
 
-RPC Refactor
+RPC
 *************
 
+Overview
+========
+
 To communicate between ``traffic_manager`` and ``traffic_server`` there is an RPC mechanism in
-``mgmt/api``. This is a simple serialization style RPC which runs over sockets. Current the logic is
-tightly embedded in the management library. It would be useful for various reasons to extract this
-in to a separate library.
+``mgmt/``. This is a simple serialization style RPC which runs over sockets.
+
+
+Serialization Mechanism
+=======================
+
+A RPC message is sent as a ``MgmtMessageHdr`` followed by the serialized data in bytes. 
+
+.. class:: MgmtMarshall
+
+   This is the class used to marshall data objects. It provides functions to marshall and unmarshall data. Each data object is associated with a field. Fields can be:
+
+.. enum:: MGMT_MARSHALL_INT
+.. enum:: MGMT_MARSHALL_LONG
+.. enum:: MGMT_MARSHALL_STRING
+.. enum:: MGMT_MARSHALL_DATA
+
+
+Marshalling:
+++++++++++++
+
+.. function:: ssize_t MgmtMarshall::mgmt_message_marshall(void *buf, size_t remain, const MgmtMarshallType *fields, unsigned count, ...)
+
+   Variable argument wrapper for ``mgmt_message_marshall_v``. Allows for different datatypes to be marshalled together as long as a field is specified for each data object.
```

Comment:
What kind of argument needs to be passed here? Pointers? References?

Reply:
References. Updated. I'll change the others too.